File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans;
        for(int i=0;i<=n;i++){
            int temp=i;
            int temp_ones = 0;
            while(temp){
                if(temp%2==1){
                    temp_ones++;
                }
                temp=temp>>1;
            }
            ans.push_back(temp_ones);
        }
        return ans;

        
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        temp = []
        ans = []

        def rec(temp,i,total):
            if total == target:
                ans.append(temp.copy())
                return
            if total > target:
                return
            if i == len(nums):
                return
            
            # same nos
            temp.append(nums[i])
            total += nums[i]
            rec(temp,i,total)

            # different nos
            temp.pop()
            total -= nums[i]
            rec(temp,i+1,total)
            
            
        rec(temp,0,0)
        return ans



        